File: python/order_status.py
"""Check what happened to an order by OID."""

import json
import sys
from client import post_endpoint, address

oid = int(sys.argv[1]) if len(sys.argv) > 1 else 0
if oid == 0:
    print("Usage: python3 order_status.py <oid>")
    print("Get OIDs from: python3 open_orders.py")
    sys.exit(1)

result = post_endpoint("/orderStatus", {"user": address, "oid": oid})

if result["status"] == "unknownOid":
    print(f"Order {oid}: not found")
    print(f"  {result['explanation']}")
    sys.exit(0)

spot = " [SPOT]" if result.get("isSpot") else ""
print(f"Order {oid} on {result['name']}{spot}: {result['status']}")
print(f"  {result['explanation']}")
if result.get("order"):
    print(f"  Details: {json.dumps(result['order'], indent=2)}")

File: typescript/src/orderStatus.ts
import { postEndpoint, address } from "./client";

async function main() {
  const oid = parseInt(process.argv[2] || "0", 10);
  if (oid === 0) {
    console.log("Usage: npx ts-node src/orderStatus.ts <oid>");
    console.log("Get OIDs from: npx ts-node src/openOrders.ts");
    process.exit(1);
  }

  const result = await postEndpoint("/orderStatus", { user: address, oid });

  if (result.status === "unknownOid") {
    console.log(`Order ${oid}: not found`);
    console.log(`  ${result.explanation}`);
    process.exit(0);
  }

  const spot = result.isSpot ? " [SPOT]" : "";
  console.log(`Order ${oid} on ${result.name}${spot}: ${result.status}`);
  console.log(`  ${result.explanation}`);
  if (result.order) {
    console.log(`  Details: ${JSON.stringify(result.order, null, 2)}`);
  }
}

main();
